src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -338,6 +343,7 @@ private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, Configu
                 if (watchedKv != null)
                 {
                     data[watchedKey] = watchedKv;
+                    _watchedSettingsCache[watchedKeyLabel] = watchedKv;

```

Comment:
In case of Select("TestKey", "Label2") and Register("TestKey", "Label1").Register("TestKey", "Label2"), this implementation would fetch the value of ("TestKey", "Label2") again even though it was retrieved using Select.

Reply:
That is correct. There is no clean way to keep track of which refresh registered key ends up in application configuration, apart from overwriting and letting the last KV win. Therefore, if there are same keys with multiple labels registered for refresh, we will fetch all of them again even if one of them was already fetched by the `Select` clause.

Reply:
Discussed offline with Jimmy. From users' point of view, every refresh registered setting will be fetched in a new server call. Avoiding some of these calls is an optimization we provide, but don't necessarily promise. In this case, the optimization would make the code would overly complicated and is not worth it.